### PR TITLE
feat(DropdownList)

### PR DESCRIPTION
**Pull request by CedricPoon-SeekAsia**
Separation of DropdrownList from JobsDB footer local component into standalone SEEK Asia component.

EXAMPLE USAGE:

```js
<DropdownList value='Description'>
  <ListItem value="Children" />
</DropdownList>
```

### CODE REVIEW

File: jobsDB/Footer/FooterHK/FooterHK.js [hunk 1]
```diff
@@ -4,13 +4,18 @@ import PropTypes from 'prop-types';
...
 import UpperFooter from './components/UpperFooter/UpperFooter';
-import { Text, PageBlock } from 'seek-asia-style-guide/react';
+import { Text, PageBlock, ListItem } from 'seek-asia-style-guide/react';
 import { getLocalization } from './localization';
 
 export const makeDefaultLinkRenderer = () => {
   const DefaultLinkRenderer = ({ href, children, ...props }) => (
     <a className={styles.link} href={href} {...props}>
-      {children}
+      {React.Children.map(children, child => {
```
Comment (mengtzu) [inline jobsDB/Footer/FooterHK/FooterHK.js]: Why `React.Children` rather than just the `children` prop?

File: react/DropdownList/DropdownList.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            className={classnames({
+              [styles.chevron]: true,
+              [styles[`${chevronAlignment}`]]: true
+            })}
+            direction={this.state.chevronDirection}
+          />
+        </div>
+        <div className={this.state.isDropdownOpen ? styles.dropdown : styles.dropdownNoShow}>
+          {
+            React.Children.map(children, child => {
```
Comment (mengtzu) [inline react/DropdownList/DropdownList.js]: This seems kind of elaborate, what's it all for?

Comment (CedricPoon-SeekAsia) [reply react/DropdownList/DropdownList.js]: will discuss with @patrickliu-sa for the behaviour of children components

File: jobsDB/Footer/FooterHK/FooterHK.js (hunk 1 above)
Comment (CedricPoon-SeekAsia) [reply jobsDB/Footer/FooterHK/FooterHK.js]: For passing ```noShadow``` to all ```<ListItem />``` child

File: react/DropdownList/DropdownList.js (hunk 2 above)
Comment (CedricPoon-SeekAsia) [reply react/DropdownList/DropdownList.js]: @mengtzu may I ask when the ```dropdownList``` contains prop ```compact```, what should it do to pass to its children?

Comment (patrickliu-sa) [reply react/DropdownList/DropdownList.js]: @mengtzu Joanne mentioned a scenario about when to use ```compact``` in a page level (which will apply to all components at once), sounds like a global ```const``` under certain criteria.

Comment (CedricPoon-SeekAsia) [reply react/DropdownList/DropdownList.js]: @mengtzu @patrickliu-sa As ```compact``` requires revision on its usage as global level over components, the adoption of ```compact``` on children components will be ignored.

**Comment by CedricPoon-SeekAsia** [comment]
@mengtzu @patrickliu-sa This PR is reopened. Only ```noShadow``` is remained to change children's prop. Feel free to drop any new comments :)

**Comment by mengtzu** [comment]
I'd say if you're going to do it this way, make the behaviour controllable by a prop and default it to not happening.  I'm a bit worried about silently modifying the behaviour of children.

**Comment by CedricPoon-SeekAsia** [comment]
@mengtzu I have removed all children modifying behaviour. Please do review ;)

File: jobsDB/Footer/FooterHK/components/PartnerSites/PartnerSites.less [hunk 3]
```diff
@@ -1,6 +1,12 @@
 @import (reference) "~seek-asia-style-guide/theme";
 @import "../UpperFooter/UpperFooter.less";
 
+.partnersList {
```
Comment (mengtzu) [inline jobsDB/Footer/FooterHK/components/PartnerSites/PartnerSites.less]: What's this for?  Normal assumption is a lot of `inherit`s wouldn't do anything?  I assume there's some other value wandering in from somewhere to make trouble?

**Pull request merged**